docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+        - [Installing root certificate in Ubuntu](#installing-root-certificate-in-ubuntu)
+        - [Installing root certificate in Centos](#installing-root-certificate-in-centos)
+    - [Testing TLS-secured plugin communication in local setup](#testing-tls-secured-plugin-communication-in-local-setup)
+        - [Starting a telemetry workflow:](#starting-a-telemetry-workflow)
+
+<!-- /TOC -->
+
+
+## Overview
+
+This document provides instructions on how to obtain certficates to evaluate TLS-secured plugin communication in Snap.
+
+Following routines are described:
+- obtaining self-signed certificates for evaluation in local test environment,
+- obtaining legitimate certificate from a certain CA authority,
+- installing root certificate in OS for validating endpoints' certificates,
+- starting telemetry workflow with secured plugin communication.
+
+### Remarks
+- This document can't be considered a definite guide to obtaining certificates. It aims only as guiding user in acquiring a certificate.
+- This guide doesn't indicate how to handle certificate revocation as that service is not yet supported in Snap.
+- Every organization has to review its requirements prior to obtaining certificates or setting up any secure communication scheme.
+
+## Obtaining self-signed certificates
+
+This section describes how to obtain certificates signed by a local CA.
+
+In the course of this guide user will establish a local CA, becoming able to establish local chains of trust. Thus the user will be able to generate certificates for Snap plugins and Snap daemon (`snapteld`) that all share common trust root.
+
+### Remarks
+
+- CA private key needs to be protected, possibly on a dedicated machine or media.
+- Obtained certificates won't be valid outside local setup.
+
+### Procedure
+
+1. Install tool [certstrap](https://github.com/square/certstrap) for generating test certificates. Further steps will assume that `certstrap` is available under `$PATH` location.
+1. Generate root CA certificate:
+    ```
+    certstrap init --cn "snaptest-ca" --o "snaptest" --ou "ca" --key-bits 2048 --years 1 --passphrase '<ca-secure-passphrase>'
+    ```
+    Relevant files:
+    * `./out/snaptest-ca.key` - CA's private key, encrypted with `<ca-secure-passphrase>`; needs to be protected from unauthorized access,
+    * `./out/snaptest-ca.crt` - CA's public certificate, (a certificate root).
+1. Generate server certificate and key to use by plugins:
+    ```
+    certstrap request-cert --cn "snaptest-srv" --ip "127.0.0.1" --domain "localhost" --passphrase '' --key-bits 2048 --o "snaptest" --ou "server"
+    certstrap sign "snaptest-srv" --CA "snaptest-ca" --passphrase '<ca-secure-passphrase>' --years 1
+    ```
+    This will produce a private key certificate file necessary to establish secure TLS channel on server. Server's private key file should not be passphrase-protected (or each plugin would have to ask for passphrase on start).
```

Comment:
This step sounds tricky. Maybe move the warning to not use a passphrase to the section above and make sure the code snippet does not create a passphrase (so that you follow your own recommendation).